jee-learning-platform/backend/app/routes/content.py

```
@@ -0,0 +1,318 @@
+from sqlalchemy import and_
+from app.database.database import get_db
+from app.models import models, schemas
+from app.utils.auth import get_current_active_user, is_teacher_or_admin
+from app.services.analytics_service import AnalyticsService
+
+router = APIRouter(prefix="/content", tags=["content"])
+
+# Course Routes
+@router.get("/courses", response_model=List[schemas.CourseNavigation])
+async def get_courses_with_navigation(
+    current_user: models.User = Depends(get_current_active_user),
```

Ruff (0.11.9)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)